bond/bond.go

```diff
@@ -198,6 +199,9 @@ func setLinksinNetNs(bondConf *bondingConfig, nspath string, releaseLinks bool)
 		if err := netNs.Set(); err != nil {
 			return fmt.Errorf("failed to enter netns %q: %v", netNs, err)
 		}
+		defer func() {
```

Comment:
why do you defer set the Ns? 
is there a specific reason for this? 
Also why ignore the error ?

Reply:
thought it would be prettier to defer the set NS because it  behaves as a cleanup function for the netNs.Set()  (returning at the end of the function the init namespace). about the error wasn't sure if we would like to return it because its an error only in the cleanup .
But i agree with you it should be returned, my mistake. 
removed the deffer and added the set ns at the end of the function